scripts/scil_compute_local_tracking.py

```diff
@@ -164,13 +207,114 @@ def tqdm_if_verbose(generator: Iterable, verbose: bool, *args, **kwargs):
+        filtered_streamlines, seeds = \
+            zip(*((s, p) for s, p in tqdm_if_verbose(
+                streamlines_generator, verbose=args.verbose,
+                total=total_nb_seeds, miniters=int(total_nb_seeds / 100))
+                  if scaled_min_length <= length(s) <= scaled_max_length))
+        data_per_streamlines = {'seeds': lambda: seeds}
+    else:
+        filtered_streamlines = \
+            (s for s in tqdm_if_verbose(
+                streamlines_generator, verbose=args.verbose,
+                total=total_nb_seeds, miniters=int(total_nb_seeds / 100))
+             if scaled_min_length <= length(s) <= scaled_max_length)
+        data_per_streamlines = {}
+
+    if args.compress:
+        # Compressing. Threshold is in mm, but we are working in voxel space.
+        # Equivalent of sft.to_voxmm:  streamline *= voxres
+        # Equivalent of sft.to_vox: streamline /= voxres
+        filtered_streamlines = (
+            compress_streamlines(s * voxel_size, args.compress) / voxel_size
+            for s in filtered_streamlines)
+
+    # IMPORTANT: streamlines are dumped in voxel space
+    tractogram = LazyTractogram(lambda: filtered_streamlines,
+                                data_per_streamlines,
+                                affine_to_rasmm=seed_img.affine)
+
+    filetype = nib.streamlines.detect_format(args.out_tractogram)
+    reference = get_reference_info(seed_img)
+    header = create_tractogram_header(filetype, *reference)
+
+    # Use generator to save the streamlines on-the-fly
+    nib.streamlines.save(tractogram, args.out_tractogram, header=header)
+
+
+def save_tractogram_gpu(streamlines_generator, odf_sh_img, args):
+    voxel_size = odf_sh_img.header.get_zooms()[0]
+
+    # wrapper for tracker.track() yielding one TractogramItem per
+    # streamline for use with the LazyTractogram.
+    def tracks_generator_wrapper():
+        for strl, seed in streamlines_generator:
+            # seeds are returned with origin `center`
+            dps = {}
+            if args.save_seeds:
+                dps['seeds'] = seed
+
+            # IMPORTANT: Streamlines are dumped in world space (mm).
+            # TODO: Investigate why the streamline must NOT be shifted to
+            # origin `center` for LazyTractogram.
+            strl *= voxel_size  # in mm.
+            if args.compress:
+                strl = compress_streamlines(strl, args.compress)
+            yield TractogramItem(strl, dps, {})
+
+    # instantiate tractogram
+    tractogram = LazyTractogram.from_data_func(tracks_generator_wrapper)
+    tractogram.affine_to_rasmm = odf_sh_img.affine
+
+    filetype = nib.streamlines.detect_format(args.out_tractogram)
+    reference = get_reference_info(odf_sh_img)
+    header = create_tractogram_header(filetype, *reference)
+
+    # Use generator to save the streamlines on-the-fly
+    nib.streamlines.save(tractogram, args.out_tractogram, header=header)
+
+
 def main():
+    t_init = perf_counter()
```

Comment:
Can you add a small in-line comment explaining what did does, considering it is not often used in our scripts?